### PR TITLE
Fix configure_earlgrey issue when either an Env Var or CommandLineArg is present in the Launch Action and not in the Test Action.

### PR DESCRIPTION
**Pull request by tirodkar**
Fixes https://github.com/google/EarlGrey/issues/221.

Issue - We have an OR check for if you have Env Variable or CommandLineArgs and your Test Action uses the vars from the launch action. However, if either one of them is empty, you can hit a nil element issue. This adds a check around that. Thanks @gurgeous

### CODE REVIEW

**Comment by bootstraponline** [comment]
We need an automated test to detect the regression. I reverted `gem/lib/earlgrey/configure_earlgrey.rb` and even with the updated `EarlGreyExampleTests.xcscheme`, all the tests still pass. This is expected because the Ruby tests are based on fixtures. Updating the Demo project scheme does nothing.

**Comment by tirodkar** [comment]
Agreed, I'll try and get the fixtures updated with a new project for this change.

**Comment by tirodkar** [comment]
Updated. Added a couple of more projects.

**Comment by bootstraponline** [comment]
I verified the tests detect the issue now. Looks good to me.

File: Demo/EarlGreyExample/EarlGreyExample.xcodeproj/xcshareddata/xcschemes/EarlGreyExampleTests.xcscheme [hunk 1]
```diff
@@ -36,6 +36,13 @@
       debugDocumentVersioning = "YES"
       debugServiceExtension = "internal"
       allowLocationSimulation = "YES">
+      <EnvironmentVariables>
+         <EnvironmentVariable
+            key = "FooKey"
```
Comment (khandpur) [inline Demo/EarlGreyExample/EarlGreyExample.xcodeproj/xcshareddata/xcschemes/EarlGreyExampleTests.xcscheme]: So we know this is for testing purposes:

FooKeyForTesting
BarValueForTesting

**Comment by khandpur** [comment]
LGTM

**Pull request merged**